external/s3-sqs/src/main/scala/org/apache/spark/sql/sqs/SqsClient.scala

```diff
@@ -0,0 +1,267 @@
+      logError("Max retries reached")
+      exception = Some(new SparkException("Unable to receive Messages from SQS for " +
+        s"${sqsMaxRetries} times Giving up. Check logs for details."))
+    } else {
+      logWarning(s"Attempt ${retriesOnFailure}." +
+        s"Will reattempt after ${sqsFetchIntervalSeconds} seconds")
+    }
+  }
+
+  private def createSqsClient(): AmazonSQS = {
+    try {
+      val isClusterOnEc2Role = hadoopConf.getBoolean(
+        "fs.s3.isClusterOnEc2Role", false) || hadoopConf.getBoolean(
+        "fs.s3n.isClusterOnEc2Role", false) || sourceOptions.useInstanceProfileCredentials
+      if (!isClusterOnEc2Role) {
+        val accessKey = hadoopConf.getTrimmed("fs.s3n.awsAccessKeyId")
+        val secretAccessKey = new String(hadoopConf.getPassword("fs.s3n.awsSecretAccessKey")).trim
+        logInfo("Using credentials from keys provided")
+        val basicAwsCredentialsProvider = new BasicAWSCredentialsProvider(
+          accessKey, secretAccessKey)
+        AmazonSQSClientBuilder
+          .standard()
+          .withClientConfiguration(new ClientConfiguration().withMaxConnections(maxConnections))
+          .withCredentials(basicAwsCredentialsProvider)
+          .withRegion(region)
+          .build()
+      } else {
+        logInfo("Using the credentials attached to the instance")
+        val instanceProfileCredentialsProvider = new InstanceProfileCredentialsProviderWithRetries()
+        AmazonSQSClientBuilder
+          .standard()
+          .withClientConfiguration(new ClientConfiguration().withMaxConnections(maxConnections))
+          .withCredentials(instanceProfileCredentialsProvider)
+          .build()
+      }
+    } catch {
+      case e: Exception =>
+        throw new SparkException(s"Error occured while creating Amazon SQS Client ${e.getMessage}")
+    }
+  }
+
+  def addToDeleteMessageQueue(messageReceiptHandles: List[String]): Unit = {
+    deleteMessageQueue.addAll(messageReceiptHandles.asJava)
+  }
+
+  def deleteMessagesFromQueue(): Unit = {
+    try {
+      var count = -1
+      val messageReceiptHandles = deleteMessageQueue.asScala.toList
+      val messageGroups = messageReceiptHandles.sliding(10, 10).toList
+      messageGroups.foreach { messageGroup =>
+        val requestEntries = messageGroup.foldLeft(List[DeleteMessageBatchRequestEntry]()) {
+          (list, messageReceiptHandle) =>
+            count = count + 1
+            list :+ new DeleteMessageBatchRequestEntry(count.toString, messageReceiptHandle)
+        }.asJava
+        val batchResult = sqsClient.deleteMessageBatch(sqsUrl, requestEntries)
+        if (!batchResult.getFailed.isEmpty) {
+          batchResult.getFailed.asScala.foreach { entry =>
+            sqsClient.deleteMessage(
+              sqsUrl, requestEntries.get(entry.getId.toInt).getReceiptHandle)
+          }
+        }
+      }
+    } catch {
+      case e: Exception =>
+        logWarning(s"Unable to delete message from SQS ${e.getMessage}")
```

Comment:
should log the exception. Also, some java exceptions (including NullPointerException) actually have a null .getMessage; the toString value is better

Reply:
will pick this change